### DISCUSSION TITLE
CSV import fails on many-to-one relationship with a unique field

### DISCUSSION THREAD

**Discussion by noahjahn** [directus/directus#16442] [Archive]
To start here are some example collections:

Collection `categories`:
* `name` - input, unique, not null, required

Collection  `books`:
* `title` - input, not null, required
* `category` - many-to-one relationship on `categories.id`

--

If I try to import a CSV with a fresh set of data with more than one book with the same category, the import fails saying that the category selected must be unique. Here is an example CSV import for to be imported on the `books` collection:

```csv
"title","category.name"
"Book title 1","Fiction"
"A different book title","Fiction"
```

Error message on the front-end:
```
This file's data structure doesn't match the collection.
```

Error message from the server:
```json
{
	"errors": [
		{
			"message": "Field \"name\" has to be unique.",
			"extensions": {
				"code": "RECORD_NOT_UNIQUE",
				"collection": "categories",
				"field": "name",
				"invalid": "test2"
			}
		}
	]
}
```

[The code appears to be using an upsert though](https://github.com/directus/directus/blob/0fc4c501c3c37a8eb22d4ac36712e2401faf5ae6/api/src/services/import-export.ts#L77), am I creating the CSV incorrectly? I'm hoping not to have to manually import each relationship first, export the ID created for that relationship and combine it with the books CSV import file, then re-import the books file.

**Comment by rijkvanzanten** [comment]
The data set you're showing will attempt to create a new category with the name "Fiction" for each of the books. In order to reuse the existing category, make sure the ID is present:

```
"title","category"
"Book title 1","1"
"A different book title","1"
```

**Reply by noahjahn** [reply]
Thanks for the quick response, I'll make this a feature request 😅
